packages/core/upload/admin/src/pages/App/index.js

```diff
@@ -1,31 +1,122 @@
 import React from 'react';
-import { Switch, Route, Redirect } from 'react-router-dom';
-import { LoadingIndicatorPage, useRBAC } from '@strapi/helper-plugin';
-import pluginId from '../../pluginId';
+import { Redirect } from 'react-router-dom';
+import { useIntl } from 'react-intl';
+import styled from 'styled-components';
+import {
+  LoadingIndicatorPage,
+  useRBAC,
+  useFocusWhenNavigate,
+  NoPermissions,
+  NoMedia,
+  AnErrorOccurred,
+} from '@strapi/helper-plugin';
+import { Layout, HeaderLayout, ContentLayout, ActionLayout } from '@strapi/parts/Layout';
+import { Main } from '@strapi/parts/Main';
+import { Button } from '@strapi/parts/Button';
+import AddIcon from '@strapi/icons/AddIcon';
+import { Box } from '@strapi/parts/Box';
+import { BaseCheckbox } from '@strapi/parts/BaseCheckbox';
+import { Search } from './components/Search';
+import { ListView } from './components/ListView';
+import { useAssets } from '../../hooks/useAssets';
+import { getTrad } from '../../utils';
 import pluginPermissions from '../../permissions';
-import { AppContext } from '../../contexts';
 
-import HomePage from '../HomePage';
+const BoxWithHeight = styled(Box)`
+  height: ${32 / 16}rem;
+  display: flex;
+  align-items: center;
+`;
 
 const App = () => {
   const state = useRBAC(pluginPermissions);
+  const { formatMessage } = useIntl();
+  const { data, isLoading, error } = useAssets({
+    skipWhen: !state.allowedActions.canMain,
+  });
 
-  // Show a loader while all permissions are being checked
-  if (state.isLoading) {
-    return <LoadingIndicatorPage />;
-  }
+  useFocusWhenNavigate();
+
+  const canRead = state.allowedActions.canMain;
+  const loading = state.isLoading || isLoading;
 
-  if (state.allowedActions.canMain) {
-    return (
-      <AppContext.Provider value={state}>
-        <Switch>
-          <Route path={`/plugins/${pluginId}`} component={HomePage} />
-        </Switch>
-      </AppContext.Provider>
-    );
+  if (!loading && !canRead) {
+    return <Redirect to="/" />;
   }
 
-  return <Redirect to="/" />;
+  return (
+    <Layout>
+      <Main aria-busy={loading}>
+        <HeaderLayout
+          title={formatMessage({
+            id: getTrad('plugin.name'),
+            defaultMessage: 'Media Library',
+          })}
+          subtitle={formatMessage(
+            {
+              id: getTrad('header.content.assets-multiple'),
```

Comment:
Here you can create a new translation that handles the singular and the plural forms directly or use both the `header.content.assets.assets-multiple` or `header.content.assets.assets-single` ones.

We will clean the translations when the migration is over.

In case you decide to create a new one:

```
"header.content.label" : "{number, plural, =0 {# assets} one {# asset} other {# assets}}"
```